// usebootstrap/dist/runtime/composables/html/useAnchor.mjs
import { computed } from "vue";
import { addProp, hasValue } from "../../utils/useProps.mjs";
import { EventEmitProps } from "../../utils/useEventEmitter.mjs";
export const AnchorProps = {
  ...EventEmitProps,
  button: {
    type: [String, Boolean]
  },
  size: {
    type: String
  },
  disabled: {
    type: Boolean
  },
  stretchedLink: {
    type: Boolean
  },
  link: {
    type: String
  },
  to: {
    type: String
  }
};
export function useAnchor(props) {
  return {
    class: computed(() => {
      return {
        "stretched-link": props.stretchedLink,
        disabled: props.disabled,
        [`link-${props.link}`]: props.link,
        btn: props.button,
        [`btn-${props.button}`]: hasValue(props.button),
        [`btn-${props.size}`]: props.size
      };
    }),
    attr: computed(() => {
      return {
        ...addProp(props.button, "role", "button"),
        ...addProp(props.disabled, "aria-disabled", "true"),
        ...addProp(props.href, "href", props.href),
        ...addProp(props.target, "target", props.target),
        ...addProp(props.to, "to", props.to)
      };
    })
  };
}
