// usebootstrap/dist/runtime/components/bootstrap/listGroup/listGroupItemAction.mjs
import { defineComponent, h, ref, computed } from "vue";
import { useBlock, BlockProps } from "../../../composables/base/useBlock.mjs";
import { hProps } from "../../../utils/useProps.mjs";
import { ActiveProps, useActive } from "../../../composables/bootstrap/useItemsActive.mjs";
import { IDProps, useID } from "../../../composables/attributes/useID.mjs";
import { ToggleProps, useToggle } from "../../../composables/bootstrap/useToggle.mjs";
import {
  AnchorProps,
  useAnchor
} from "../../../composables/html/useAnchor.mjs";
import { useScrollSpyItem } from "../../../composables/bootstrap/useScrollSpyItem.mjs";
import BsLink from "../../nuxt/bslink.mjs";
export default defineComponent({
  name: "BsListItem",
  props: {
    ...AnchorProps,
    ...BlockProps,
    ...ActiveProps,
    ...ToggleProps,
    ...IDProps,
    theme: {
      type: String,
      default: void 0
    }
  },
  setup(props, context) {
    const elementRef = ref();
    const Anchor = useAnchor(props);
    const block = useBlock(props);
    const id = useID(props, "list-item");
    const toggle = useToggle(props, elementRef);
    const active = useActive(props, "list", elementRef);
    const spyItem = useScrollSpyItem(props, elementRef);
    const current = {
      class: computed(() => {
        return {
          "list-group-item": true,
          [`list-group-item-${props.theme}`]: props.theme
        };
      }),
      ref: elementRef
    };
    return () => h(
      BsLink,
      hProps(current, id, Anchor, block, active, toggle, spyItem),
      context.slots
    );
  }
});
