// usebootstrap/dist/runtime/components/bootstrap/modal/modalToggleButton.mjs
import { defineComponent, h, ref } from "vue";
import { hProps } from "../../../utils/useProps.mjs";
import { BlockProps, useBlock } from "../../../composables/base/useBlock.mjs";
import {
  ButtonProps,
  useButton
} from "../../../composables/html/useButton.mjs";
import { ToggleProps, useToggle } from "../../../composables/bootstrap/useToggle.mjs";
export default defineComponent({
  name: "BsModalToggleButton",
  props: {
    ...BlockProps,
    ...ButtonProps,
    ...ToggleProps,
    toggle: {
      type: String,
      default: "modal"
    },
    type: {
      type: String,
      default: "button"
    }
  },
  setup(props, context) {
    const elementRef = ref();
    const block = useBlock(props);
    const Button = useButton(props);
    const toggle = useToggle(props, elementRef);
    const current = {
      attr: {
        type: props.type
      },
      ref: elementRef
    };
    return () => h(
      "button",
      hProps(current, Button, block, toggle),
      context.slots
    );
  }
});
